component/sidetree/option/create/option.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package create implements sidetree create did option
+//
+package create
+
+import (
+	"crypto"
+
+	docdid "github.com/hyperledger/aries-framework-go/pkg/doc/did"
+
+	"github.com/hyperledger/aries-framework-go-ext/component/sidetree/doc"
+)
+
+// Opts create did opts.
+type Opts struct {
+	PublicKeys         []doc.PublicKey
+	Services           []docdid.Service
+	SidetreeEndpoints  []string
+	RecoveryPublicKey  crypto.PublicKey
+	UpdatePublicKey    crypto.PublicKey
+	SigningKey         crypto.PrivateKey
+	SigningKeyID       string
+	MultiHashAlgorithm uint
+}
+
+// Option is a create DID option.
+type Option func(opts *Opts)
+
+// WithPublicKey add DID public key.
+func WithPublicKey(publicKey *doc.PublicKey) Option {
+	return func(opts *Opts) {
+		opts.PublicKeys = append(opts.PublicKeys, *publicKey)
+	}
+}
+
+// WithService add service.
+func WithService(service *docdid.Service) Option {
+	return func(opts *Opts) {
+		opts.Services = append(opts.Services, *service)
+	}
+}
+
+// WithSidetreeEndpoint sidetree url.
+func WithSidetreeEndpoint(sidetreeEndpoint string) Option {
```

Comment:
We should also have a way to inject an endpoint discovery mechanism.

Reply:
done